funcs_scripts.php

```diff
@@ -123,6 +124,12 @@ function rename_file_if_exists($oldFilename, $newFilename)
  */
 function module_is_recipe()
 {
+    global $GITHUB_REF;
+    $repoData = MetaData::getMetaDataForRepository($GITHUB_REF);
+    if (isset($repoData['type'])) {
+        return $repoData['type'] === 'recipe';
+    }
```

Comment:
The `type` key is only available for repos in the "supportedModules" category - but e.g. testing-recipe is a recipe so we need to still check the old way as well.

Same idea with all other changes below checking `type`.

Reply:
Why do we even need to use the supported-modules repositories.json as the source of truth instead of composer.json of each module? Nothing was broken before?

Ditto for all the other updates in the PR doing the same e.g. is_module()

Reply:
Just feels more correct. I'll remove it if you insist, but I don't see any reason to.

Reply:
> looks just like we don't trust supported-modules to be correct

A comment could easily solve that lol. I'll remove it just to avoid ping pong, I think either way the result ends up being the same so it doesn't matter.

Reply:
done